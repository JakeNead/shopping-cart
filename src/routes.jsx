import Root from "./components/root/Root";
import Home from "./components/home/Home";
import Shop from "./components/shop/Shop";
import Cart from "./components/cart/Cart";
import About from "./components/about/About";
import Error from "./components/Error";
import { createBrowserRouter } from "react-router-dom";

const routes = createBrowserRouter([
  {
    path: "/",
    element: <Root />,
    errorElement: <Error />,
    children: [
      { index: true, element: <Home /> },
      { path: "/shop", element: <Shop /> },
      { path: "/cart", element: <Cart /> },
      { path: "/about", element: <About /> },
    ],
  },
]);

export default routes;
