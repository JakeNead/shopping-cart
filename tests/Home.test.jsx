import routes from "../src/routes";
import userEvent from "@testing-library/user-event";
import { render, screen } from "@testing-library/react";
import { RouterProvider } from "react-router-dom";

describe("Home component", () => {
  it("Renders home page content", () => {
    const { container } = render(<RouterProvider router={routes} />);
    expect(container).toMatchSnapshot();
  });

  it("Routes to shop component from 'Shop Now' button", async () => {
    const user = userEvent.setup();
    render(<RouterProvider router={routes} />);
    const button = screen.getByRole("button", { name: "Shop Now" });

    await user.click(button);
    expect(screen.getByText(/Loading.../)).toBeInTheDocument();
  });
});
