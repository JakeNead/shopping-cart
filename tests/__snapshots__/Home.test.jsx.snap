// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home component > Renders home page content 1`] = `
<div>
  <header>
    <nav>
      <h1>
        FakeStore
      </h1>
      <ul
        class="navLinks"
      >
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            class=""
            href="/shop"
          >
            Shop
          </a>
        </li>
        <li>
          <a
            class=""
            href="/about"
          >
            About Us
          </a>
        </li>
      </ul>
      <a
        class="cartIcon"
        href="/cart"
      >
        <svg
          height="24"
          viewBox="0 -960 960 960"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M280-80q-33 0-56.5-23.5T200-160q0-33 23.5-56.5T280-240q33 0 56.5 23.5T360-160q0 33-23.5 56.5T280-80Zm400 0q-33 0-56.5-23.5T600-160q0-33 23.5-56.5T680-240q33 0 56.5 23.5T760-160q0 33-23.5 56.5T680-80ZM246-720l96 200h280l110-200H246Zm-38-80h590q23 0 35 20.5t1 41.5L692-482q-11 20-29.5 31T622-440H324l-44 80h480v80H280q-45 0-68-39.5t-2-78.5l54-98-144-304H40v-80h130l38 80Zm134 280h280-280Z"
          />
        </svg>
        <div
          class="cartTotal"
        >
          <b>
            0
          </b>
           items
        </div>
      </a>
    </nav>
  </header>
  <main
    class="_homeMain_b0cb5b"
  >
    <section
      class="_heroContainer_b0cb5b"
    >
      <div
        class="_heroText_b0cb5b"
      >
        <h1>
          Welcome to Our FakeStore
        </h1>
        <p>
          Discover the latest trends in clothing, electronics, and jewelry!
        </p>
        <a
          href="/shop"
        >
          <button
            class="_heroButton_b0cb5b"
          >
            Shop Now
          </button>
        </a>
      </div>
    </section>
    <section
      class="_freeShipping_b0cb5b"
    >
      <h2>
        Free shipping on orders over $50
      </h2>
      <p>
        Whether you're stocking up on essentials or treating yourself to something special, enjoy our free shipping on us with orders over $50! Shop with us today and take advantage of our free shipping offer to make your online shopping experience even more rewarding!
      </p>
    </section>
  </main>
  <footer>
    <h4>
      Made by Jake
    </h4>
    <p>
      Copyright Â© 2024
    </p>
    <div
      class="links"
    >
      <a
        href="https://github.com/JakeNead"
      >
        <svg
          class="bi bi-github"
          fill="currentColor"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27s1.36.09 2 .27c1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.01 8.01 0 0 0 16 8c0-4.42-3.58-8-8-8"
          />
        </svg>
      </a>
      <a
        href="https://www.linkedin.com/in/jakenead/"
      >
        <svg
          class="bi bi-linkedin"
          fill="currentColor"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 1.146C0 .513.526 0 1.175 0h13.65C15.474 0 16 .513 16 1.146v13.708c0 .633-.526 1.146-1.175 1.146H1.175C.526 16 0 15.487 0 14.854zm4.943 12.248V6.169H2.542v7.225zm-1.2-8.212c.837 0 1.358-.554 1.358-1.248-.015-.709-.52-1.248-1.342-1.248S2.4 3.226 2.4 3.934c0 .694.521 1.248 1.327 1.248zm4.908 8.212V9.359c0-.216.016-.432.08-.586.173-.431.568-.878 1.232-.878.869 0 1.216.662 1.216 1.634v3.865h2.401V9.25c0-2.22-1.184-3.252-2.764-3.252-1.274 0-1.845.7-2.165 1.193v.025h-.016l.016-.025V6.169h-2.4c.03.678 0 7.225 0 7.225z"
          />
        </svg>
      </a>
    </div>
  </footer>
</div>
`;
